int32 Sp1
int32 Sp2
bool Sp1_moving
bool Sp2_moving
